TypeScriptSyntaxVisualizer/TextViewCreationListener.cs

```diff
@@ -26,11 +26,22 @@ private void Caret_PositionChanged(object sender, CaretPositionChangedEventArgs
 
                 //We roll the dice on an OOM exception... :(
                 string rawText = caret.ContainingTextViewLine.Snapshot.GetText();
+                string textbeforecaret = rawText.Substring(0, position);
```

Comment:
use camelCase and verbose names for variables, for example `textBeforeCaret` and `lineNumber`